core/src/main/java/org/elasticsearch/search/aggregations/metrics/tophits/TopHitsAggregator.java

```diff
@@ -175,6 +175,12 @@ public InternalAggregation buildAggregation(long owningBucketOrdinal) {
                     FieldDoc fieldDoc = (FieldDoc) scoreDoc;
                     searchHitFields.sortValues(fieldDoc.fields, subSearchContext.sort().formats);
                 }
+                if (searchHitFields.getNestedIdentity() != null) {
+                    // In case of nested inner search hits and top hits including the index name and id is
```

Comment:
What confused me a bit about this comment is that it first sounded like we need to incluse index/id in the nested Identity, but with flag is really there to give a hint about the xContent output it seems. It feels a bit weird to me that this flag introduces some kind of "output" directive to NestedIdentity about fields that are not really part of itself but the surrounding SearchHit, or did I misunderstand something?

Reply:
No, what you write is correct. I didn't find a better way of indicating whether `SearchHit`  should serialise the _index and _id fields.

Another approach is to always serialise the _index and _id field in the case of nested search hits. This would make the code cleaner, but the _index and _id are then also included when nested search hit is an inner hit of a nested parent document and in that case the _index and _id will be serialized multiple times. However I think this might be an ok tradeoff as the _id and _index fields are only a small part of the actual search hit.

Reply:
I agree, I would lean towards always including _id/_index then, but I don't know which discussion led to removing the redundancy in the first place. Maybe it was something that people really liked to have to reduce the overall response size.

Reply:
The issue that was raised that caused the _id and _index to be removed was: #18091
This issue is more about the fact that the _index / _id wasn't always available, so it was more about dealing with an inconsistency than removing these attributes. I think with this in mind it is ok to return _id and _index attributes to nested search hits. (as long as these attributes are always serialized for nested search hits)

Reply:
@cbuescher We just discussed and we are ok with bringing back the _index and _id fields for any nested being returned irregardless its context (top_hits or inner hits)